### PR TITLE
cephadm: add config required for ganesha grace

### PR DESCRIPTION
**Pull request by mgfritch**
adds minimal config for ganesha-rados-grace

```
NFS_CORE_PARAM {{
        Enable_NLM = false;
        Enable_RQUOTA = false;
        Protocols = 4;
}}

NFSv4 {{
        Delegations = false;
        RecoveryBackend = 'rados_cluster';
        Minor_Versions = 1, 2;
}}

RADOS_KV {{
        UserId = "{user}";
        nodeid = "{nodeid}";
        pool = "{pool}";
        namespace = "{namespace}";
}}

RADOS_URLS {{
        UserId = "{user}";
        watch_url = "{url}";
}}

%url    {url}
```

Signed-off-by: Michael Fritch <mfritch@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by mgfritch** [comment]
`CACHEINODE` and `EXPORT_DEFAULTS` blocks were omitting with the assumption that the new logic defined by #32297 would want to define these in the RADOS "common config" ?

This also relates a similar issue in rook: https://github.com/rook/rook/issues/5122

**Comment by jtlayton** [comment]
Looks reasonable. I'll leave it to you and @varshar16  to settle who handles CACHEINODE and EXPORT_DEFAULTS blocks.

**Comment by varshar16** [comment]
> `CACHEINODE` and `EXPORT_DEFAULTS` blocks were omitted with the assumption that the new logic defined by #32297 would want to write these into the RADOS "common config" ?
> 
I would prefer cephadm to define these blocks and also have an option to overwrite them.

File: src/pybind/mgr/cephadm/nfs.py [hunk 1]
```diff
@@ -22,6 +22,10 @@ def __init__(self,
         self.daemon_id = daemon_id
         self.spec = spec
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/nfs.py:23]: can we assert that `self.daemon_id.startswith(self.spec.service_name)`?

File: src/pybind/mgr/cephadm/nfs.py [hunk 2]
```diff
@@ -107,20 +111,43 @@ def create_rados_config_obj(self, clobber=False):
     def get_ganesha_conf(self):
         # type: () -> str
         return '''# generated by cephadm
+NFS_CORE_PARAM {{
+        Enable_NLM = false;
+        Enable_RQUOTA = false;
+        Protocols = 4;
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/nfs.py:117]: is this something that cephadm should control, or @varshar16 ?

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2020-04-07_09:41:02-rados-wip-kefu-testing-2020-04-07-1603-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-04-07_15:13:18-rados-wip-kefu-testing-2020-04-07-1603-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-04-08_01:39:26-rados-wip-kefu-testing-2020-04-07-1603-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-04-08_03:18:15-rados-wip-kefu-testing-2020-04-07-1603-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-04-08_04:37:42-rados-wip-kefu-testing-2020-04-07-1603-distro-basic-smithi/

**Comment by mgfritch** [comment]
> Looks reasonable. I'll leave it to you and @varshar16 to settle who handles CACHEINODE and EXPORT_DEFAULTS blocks.

Do these blocks both apply equally to cephfs and RGW? It appears the `CACHEINODE` block only applies to cephfs and becomes a noop for RGW?

File: src/pybind/mgr/cephadm/nfs.py (hunk 2 above)
Comment (mgfritch) [reply src/pybind/mgr/cephadm/nfs.py:106]: It's required for nfs grace in the the bootstrap config ... we need to investigate options for allowing this to be overwritten by consumers of the orchestrator interface (rook/rook#5122)

**Comment by mgfritch** [comment]
> > `CACHEINODE` and `EXPORT_DEFAULTS` blocks were omitted with the assumption that the new logic defined by #32297 would want to write these into the RADOS "common config" ?
> 
> I would prefer cephadm to define these blocks and also have an option to overwrite them.

Added the following blocks to the default cephadm config:
```
CACHEINODE {{
        Dir_Chunk = 0;
        NParts = 1;
        Cache_Size = 1;
}}

EXPORT_DEFAULTS {{
        Attr_Expiration_Time = 0;
}}
```

Let's looks at ways to allow this to be overwritten in a future PR that applies for both rook and cephadm (rook/rook#5122)

**Comment by mgfritch** [comment]
http://pulpito.ceph.com/mgfritch-2020-04-10_16:16:11-rados-wip-mgfritch-testing-2020-04-10-0823-distro-basic-smithi/

failures related to https://github.com/ceph/ceph/pull/34443:
```
2020-04-10T17:58:23.992 INFO:tasks.workunit.client.0.smithi063.stderr:INFO:cephadm:ceph:stderr Trying to pull quay.io/ceph-ci/ceph:octopus...
2020-04-10T17:58:23.993 INFO:tasks.workunit.client.0.smithi063.stderr:INFO:cephadm:ceph:stderr   unknown: Tag octopus was deleted or has expired. To pull, revive via time machine
```

**Pull request merged**